pkg/controller/backend/backend_controller.go

```diff
@@ -150,13 +150,41 @@ func (r *ReconcileBackend) reconcile(backendResource *capabilitiesv1beta1.Backen
 		return reconcile.Result{Requeue: true}, nil
 	}
 
-	backendAPI, specErr := r.reconcileSpec(backendResource)
+	err := r.validateSpec(backendResource)
+	if err != nil {
+		if helper.IsInvalidSpecError(err) {
+			// On Validation error, no need to retry as spec is not valid and needs to be changed
+			logger.Info("ERROR", "spec validation error", err)
+			r.EventRecorder().Eventf(backendResource, corev1.EventTypeWarning, "Invalid Backend Spec", "%v", err)
+			return reconcile.Result{}, nil
+		}
+
+		return reconcile.Result{}, err
+	}
+
+	providerAccount, err := helper.LookupProviderAccount(r.Client(), backendResource.Namespace, backendResource.Spec.ProviderAccountRef, r.Logger())
```

Comment:
Code from reconcileSpec has been copied here and slightly modified.
I think we should remove reconcileSpec or update it to the new content and call it from here. Same would apply in product controller

Reply:
removed unused method